### DISCUSSION TITLE
[Scenario Editor] Empty magic book on the adventure map

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#8753] [General]
I'm not a big fan of the possibility of placing an empty magic book on the adventure map.
First of all, a magic book without a spell is not very useful.

Also, one of the problems this can cause is how to capture (remove from the adventure map, for example, to free the passage) a magic book if all the barbarian and knights heroes already have a magic book in their inventory?

My suggestion is that the magic book should contain at least one spell (the scenario creator can therefore assign several spells to the magic book) when positioned on the adventure map.

For the operation, when a hero without a magic book captures the magic book from the adventure map, he gets the magic book.
If the hero already has a magic book, the two magic books are "merged" and the spells contained in the magic book from the adventure map are added to the magic book the hero already has.
The magic book artifact on the adventure map becomes a kind of super spell scroll.

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu 
I think this is up to map maker, what spells should hero acquire. If he deliberately created a hero without magic book, then he has a idea, what type of spells will this hero gain according to the plot. Placing a default spell in the book excludes the possibility to create a hero without the particular default skill. Let's say we want to create a sorceress hero without "Bless" in her book. With the proposed logic it's not possible to create one. In my personal opinion, magic book laying on the adventure map is just a common object, that should provide same effect for all heroes picking it up. So a book with 0 spells in it is more simple and straightforward event, rather than a book, that gives different spell depending on the hero.
Players also would be confused, why sorceress hero has a spell, when picking up a book, but picking the book with knight/barbarian hero gives them 0 spells in it.
After all, If we want to make such situation, when the hero picks the book with a certain spell, we can just place a scroll or magic shrine with a particular spell in it.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

Nothing will prevents the scenario creator from choosing which spell to assign to the magic book.
The logic would be this one:
When the scenario creator adds a magic book to the adventure map, he would be prompted to assign at least one spell to the magic book.
In the hero's screen, a book is already present with the default spell for the four types of hero of magic, but nothing would prevent the magic book from being removed from the hero's inventory altogether.
The scenario creator could also add a spell to the magic book and remove the hero type's default spell.
If the scenario creator wants a hero without a spell, he doesn't give him a magic book.
Nothing is imposed except that a magic book must contain at least one spell.
This makes sense for a magic book, otherwise the empty book is just a place to prop up a wobbly piece of furniture.

Related discussion: https://github.com/ihhub/fheroes2/discussions/8755.
Related issue: https://github.com/ihhub/fheroes2/issues/8754.

**Reply by minuspinus** [reply]
The book of magic can be given through an event.

**Comment by LeHerosInconnu** [comment]
Let's go with an empty magic book on the adventure map.
I close the discussion.
